core/src/main/scala/org/apache/spark/TaskContext.scala

```diff
@@ -183,6 +183,17 @@ abstract class TaskContext extends Serializable {
   @Evolving
   def resources(): Map[String, ResourceInformation]
 
+  /**
+   * The shuffle generation ID of the stage that this task belongs to, it returns the stage
+   * attempt number while the stage is not determinate and returns none on the contrary.
+   */
+  def getShuffleGenerationId(shuffleId: Int): Option[Int] = {
```

Comment:
some places we use `-1` to indicate invalid shuffle generation id, some places we use `None`. Can we unify it?

Reply:
Yeah, thanks for the advice, compared with using Option in Java code, I change all places to use -1. Done in 5a4b8b8.